build.rs

```diff
@@ -85,24 +93,39 @@ fn zip_template_and_build_dylint_driver(manifest_dir: PathBuf, out_dir: PathBuf)
     // > * Any sub-packages will be skipped (any subdirectory that contains a Cargo.toml file).
     //
     // (from https://doc.rust-lang.org/cargo/reference/manifest.html#the-exclude-and-include-fields)
-    std::fs::rename(
-        ink_dylint_driver_dir.join("_Cargo.toml"),
-        ink_dylint_driver_dir.join("Cargo.toml"),
-    )?;
-
-    let res = build_and_zip_dylint_driver(
-        ink_dylint_driver_dir.clone(),
-        out_dir,
-        dylint_driver_dst_file,
-    );
+    let original_name = ink_dylint_driver_dir.join("_Cargo.toml");
+    if !original_name.exists() {
+        anyhow::bail!(
+            "'{:?}' does not exist, does the folder '{:?}'? {:?}",
+            original_name,
+            ink_dylint_driver_dir,
+            ink_dylint_driver_dir.exists()
```

Comment:
I guess this will always be true because `canonicalize` would fail otherwise.

Reply:
https://doc.rust-lang.org/std/fs/fn.canonicalize.html#errors
>This function will return an error in the following situations, but is not limited to just these cases:
> * path does not exist.
> * A non-final component in path is not a directory.

So it could also be that the path exists, but something else doesn't work out.

Reply:
Right so `ink_dylint_driver_dir.exists()` will always be true if we get this far?

Reply:
Ahh right, I just now checked the context, thought you annotated another line.